apps/meteor/tests/end-to-end/api/livechat/00-rooms.ts

```diff
@@ -464,6 +476,23 @@ describe('LIVECHAT - rooms', function () {
 
 			await request.get(api('livechat/room.join')).set(credentials).query({ roomId: room._id }).send().expect(200);
 		});
+		it('should allow managers to join a room which is already being served by an agent', async () => {
+			await updateSetting('Livechat_Routing_Method', 'Manual_Selection');
+			// delay for 1 second to make sure the routing queue gets stopped
+			await sleep(1000);
```

Comment:
Avoid sleeping inside tests as much as possible, unless this is a functionality requirement. This slows down tests and can cause flaky tests if more time is needed.

Reply:
The sleep when changing routing methods is for the queue to start processing or to stop doing it. Not ideal, but we have to deal with it while we have the arcane queue we have for omni.